eng/pipelines/templates/stages/archetype-sdk-bom.yml

```diff
@@ -0,0 +1,23 @@
+parameters:
+  Artifacts: []
+  SDKType: bom
+  ServiceDirectory: not-specified
+
+stages:
+  - stage: Build
+    jobs:
+      - template: ../jobs/archetype-sdk-bom.yml
+        parameters:
+          ServiceDirectory: ${{parameters.ServiceDirectory}}
+          SDKType: ${{parameters.SDKType}}
+          Artifacts: ${{parameters.Artifacts}}
+
+  # The Prerelease and Release stages are conditioned on whether we are building a pull request and the branch.
+  - ${{if and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal'))}}:
```

Comment:
As we chatted I suspect this will not work as is. I'll defer to @mitchdenny  to help deep dive into this and figure out what needs to happen.

Reply:
The POM publishing should work, and I think that signing will be a no-op. There could be some things we need to react to in the release legs of the pipeline such as version bump logic.

Reply:
My proposal is that given this won't break any current release pipelines is that we get it in and start tweaking things.